client/src/_reducers/authReducer.js

```diff
@@ -0,0 +1,51 @@
+import dayjs from 'dayjs'
+import jwt_decode from 'jwt-decode'
+import { createSlice } from '@reduxjs/toolkit'
+
+/*
+  The use of .format on dayjs returns the ISO8601 string.
+  This is an arbitrary format decision; Redux Toolkit won't store
+  non-serializable data, so we need to turn it into SOME string
+*/
+
+const initialAuthState = () => {
+  const token = localStorage.getItem('pie-token')
+  const expiration = localStorage.getItem('pie-expiration')
+  return {
+    token,
+    expiration: expiration || dayjs().format()
+  }
+}
+
+const auth = createSlice({
+  name: 'auth',
+  initialState: initialAuthState(),
+  reducers: {
+    addAuth(state, action) {
```

Comment:
cc @cjhaddad check out this sweet Redux toolkit thing; createSlice lets you define reducers and actions in one place 💘   Gonna merge this because it's getting real stale, but we can change the implementation later if you want.